message.go

```diff
@@ -1,19 +1,30 @@
 package stream_chat
 
 import (
-	"encoding/json"
 	"errors"
 	"net/http"
+	"net/url"
+	"path"
 	"time"
 )
 
+type messageType string
+
+const (
+	MessageTypeRegular   messageType = "regular"
+	MessageTypeError     messageType = "error"
+	MessageTypeReply     messageType = "reply"
+	MessageTypeSystem    messageType = "system"
+	MessageTypeEphemeral messageType = "ephemeral"
+)
+
 type Message struct {
 	ID string `json:"id"`
 
 	Text string `json:"text"`
 	HTML string `json:"html"`
 
-	Type messageType `json:"type"`
+	Type messageType `json:"type"` // one of MessageType* constants
```

Comment:
Export MessageType.

Reply:
why? it should be one of predefined constants, no other options. Sadly Go doesn't have enums

Reply:
Now it becomes impossible to use anything other than the literals. Is that the intention? Are users not allowed to use these values in their structs?